File: rdr2_banco.js
const { Low } = require('lowdb')
const { JSONFile } = require('lowdb/node')

const file = './rdr2-personagens.json'
const adapter = new JSONFile(file)
const db = new Low(adapter, { gangue: { van_der_linde: [] } })

async function main() {
    await db.read()
    
    // Arthur Morgan (Create)
    db.data.gangue.van_der_linde.push({ 
        id: 1,
        nome: "Arthur Morgan",
        idade: 36,
        sexo: "masculino",
        personalidade: "leal, corajoso, complexo",
        historia: "Protagonista leal da gangue Van der Linde, moral complexa.",
        habilidades: ["olho da morte", "natação", "tiro preciso", "sobrevivência"],
        fraquezas: ["drogas", "saúde", "conflitos internos"]
    })
    await db.write()
    
    console.log('=== BANCO RDR2 ===')
    console.log('Gangue completa:', db.data.gangue.van_der_linde)
    
    // John Marston (Create/Update)
    db.data.gangue.van_der_linde.push({
        id: 3,
        nome: "John Marston",
        idade: 38,
        sexo: "masculino",
        personalidade: "determinado, leal",
        historia: "Ex-membro buscando redenção familiar.",
        habilidades: ["tiro", "equitação", "combate"],
        fraquezas: ["impulsividade", "passado"]
    })
    await db.write()
    
    // Filter protagonistas
    const protagonistas = db.data.gangue.van_der_linde.filter(p => 
        p.personalidade.includes('leal')
    )
    console.log('Protagonistas leais:', protagonistas)
    
    // Update Arthur (tuberculose)
    const arthur = db.data.gangue.van_der_linde.find(p => p.id === 1)
    if (arthur) {
        arthur.fraquezas.push('tuberculose')
        await db.write()
        console.log('Arthur evoluiu:', arthur.fraquezas)
    }
    
    // Delete John (id 3)
    db.data.gangue.van_der_linde = db.data.gangue.van_der_linde.filter(p => p.id !== 3)
    await db.write()
    
    console.log('Gangue final:', db.data.gangue.van_der_linde)
    console.log('✅ Dados salvos em rdr2-personagens.json!')
}

main().catch(err => console.error('Erro:', err))

File: rdr2_banco.py
import json
import os

ARQUIVO = 'rdr2-personagens.json'

def carregar_dados():
    if os.path.exists(ARQUIVO):
        with open(ARQUIVO, 'r', encoding='utf-8') as f:
            return json.load(f)
    return {"gangue": {"van_der_linde": []}}

def salvar_dados(dados):
    with open(ARQUIVO, 'w', encoding='utf-8') as f:
        json.dump(dados, f, indent=4, ensure_ascii=False)

dados = carregar_dados()

# Create: adicionar Lenny
dados["gangue"]["van_der_linde"].append({
    "id": 11,
    "nome": "Lenny",
    "idade": 25,
    "sexo": "masculino",
    "personalidade": "calmo, leal, habilidoso",
    "historia": "Lenny é um membro da gangue Van der Linde que é conhecido por sua calma e habilidades de caça e sobrevivência. Ele é leal à gangue e muitas vezes atua como um caçador e rastreador.",
    "habilidades": ["caça", "rastreamento", "resistência física", "sobrevivência", "combate corpo a corpo"],
    "fraquezas": ["dificuldade em lidar com conflitos violentos", "tendência a ser reservado", "dificuldade em confiar nos outros"]
})
salvar_dados(dados)

# Read: listar membros da gangue
for p in dados["gangue"]["van_der_linde"]:
    print(f"ID: {p['id']}, Nome: {p['nome']}, Idade: {p['idade']}, Sexo: {p['sexo']}")

# Update: adicionar tuberculose à fraqueza de Arthur
for p in dados["gangue"]["van_der_linde"]:
    if p['nome'] == 'Arthur':
        p['fraquezas'].append('tuberculose')
salvar_dados(dados)

# Delete: remover membro da gangue
dados["gangue"]["van_der_linde"] = [p for p in dados["gangue"]["van_der_linde"] if p['nome'] != 'Sadie Adler']
salvar_dados(dados)